distributed/shuffle/_shuffle.py

```diff
@@ -277,6 +283,8 @@ def split_by_worker(
 
     from dask.dataframe.dispatch import to_pyarrow_table_dispatch
 
+    df = df.astype(meta.dtypes, copy=False)
```

Comment:
@jrbourbeau: Without dask/dask#10462, this line would break things. `_partitions` is meant to be an `int64`, but it's a `float` in `meta` because of an index mismatch when assigning. This causes subtle errors in P2P downstream, and things will break down. Namely, casting `_partitions` to `float` will cause filename mismatched and we read no data on `shuffle_unpack`.